docs/proposal/2022-11-22-botkube-plugins-integration.md

```diff
@@ -0,0 +1,498 @@
+
+```yaml
+entries:
+  - name: "kubectl"
+    type: "executor"
+    description: "Kubectl executor plugin."
+    version: "v1.0.0"
+    links:
+      - https://github.com/kubeshop/botkube/releases/download/v0.17.0/executor_kubectl-darwin-amd64
+      - https://github.com/kubeshop/botkube/releases/download/v0.17.0/executor_kubectl-darwin-arm64
+      - https://github.com/kubeshop/botkube/releases/download/v0.17.0/executor_kubectl-linux-amd64
+      - https://github.com/kubeshop/botkube/releases/download/v0.17.0/executor_kubectl-linux-arm64
+```
+
+In the first phase:
+
+- we support only Linux and macOS. Support for more platforms will be added when explicitly requested.
+- we support only HTTP servers. Support for different protocols will be added when explicit requested. We can utilize the [`hashicorp/go-getter`](https://github.com/hashicorp/go-getter) library to introduce support for downloading plugins directly from Git, Mercurial, Amazon S3, Google GCP. With such approach we will simply support a new syntax, e.g. for Git it will be `github.com/kubeshop/botkube-plugins?ref={tag/commit/branch}`
+
+### Releasing Botkube plugins
+
+As a part of 0.17 release, such binaries should be added to the GitHub release assets:
+
+- `plugins-index.yaml`
+- `source_kubernetes-darwin-amd64`
+- `source_kubernetes-linux-amd64`
+- `executor_kubectl-darwin-amd64`
+- `executor_kubectl-linux-amd64`
+
+Where the `plugins-index.yaml` is defined as follows:
+
+```yaml
+entries:
+  - name: "kubectl"
+    type: "executor"
+    description: "Kubectl executor plugin."
+    version: "v1.0.0"
+    links:
+      - https://github.com/kubeshop/botkube/releases/download/v0.17.0/executor_kubectl-darwin-amd64
+      - https://github.com/kubeshop/botkube/releases/download/v0.17.0/executor_kubectl-darwin-arm64
+      - https://github.com/kubeshop/botkube/releases/download/v0.17.0/executor_kubectl-linux-amd64
+      - https://github.com/kubeshop/botkube/releases/download/v0.17.0/executor_kubectl-linux-arm64
+  - name: "kubernetes"
+    type: "source"
+    description: "Kubernetes source plugin."
+    version: "v1.0.0"
+    links:
+      - https://github.com/kubeshop/botkube/releases/download/v0.17.0/source_kubernetes-darwin-amd64
+      - https://github.com/kubeshop/botkube/releases/download/v0.17.0/source_kubernetes-darwin-arm64
+      - https://github.com/kubeshop/botkube/releases/download/v0.17.0/source_kubernetes-linux-amd64
+      - https://github.com/kubeshop/botkube/releases/download/v0.17.0/source_kubernetes-linux-arm64
+```
+
+The index file can be generated automatically just by scanning directory where plugins binaries are stored.
+
+## Implementation details
+
+### E2E testing
+
+Botkube fetches the plugins index and binaries when it's started. However, if they are already present, Botkube doesn't take any action. We could leverage that and simply built all plugins binaries and include them in Docker image with in a proper directories.
```

Comment:
I tested that approach both locally and on CI. It works properly, see: https://github.com/mszostok/botkube/actions/runs/3523896348/jobs/5908734163